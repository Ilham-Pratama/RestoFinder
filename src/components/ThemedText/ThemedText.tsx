import { useTheme } from '@react-navigation/native';
import React, { useMemo } from 'react';
import { Text, TextProps } from 'react-native';

interface IThemedText extends TextProps {
  color?: string;
  darkModeColor?: string;
}

const ThemedText = (props: IThemedText) => {
  const theme = useTheme();

  const textColor = useMemo(
    () =>
      theme.dark
        ? props.darkModeColor || theme.colors.text
        : props.color || theme.colors.text,
    [props, theme]
  );

  return <Text {...props} style={[props.style, { color: textColor }]} />;
};

export default ThemedText;
