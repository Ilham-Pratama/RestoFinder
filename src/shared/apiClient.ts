import axios from 'axios';
import { YELP_API_TOKEN } from 'react-native-dotenv';

export const BASE_API_URL = 'https://api.yelp.com/v3';

export const apiClient = axios.create({
  baseURL: BASE_API_URL,
  timeout: 1000,
  headers: {
    Authorization: `Bearer ${YELP_API_TOKEN}`
  }
});
