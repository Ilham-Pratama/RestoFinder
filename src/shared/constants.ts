import { ICategoryItem } from 'components/CategoryItem/types';

export const MENU_CATEGORIES: ICategoryItem[] = [
  {
    label: 'Burger',
    value: 'burgers',
    image: require('assets/images/burger.png')
  },
  {
    label: 'Dessert',
    value: 'desserts',
    image: require('assets/images/cake.png')
  },
  {
    label: 'Pasta',
    value: 'pastashops',
    image: require('assets/images/pasta.png')
  },
  {
    label: 'Pizza',
    value: 'pizza',
    image: require('assets/images/pizza.png')
  },
  {
    label: 'Smoothies',
    value: 'juicebars',
    image: require('assets/images/smoothies.png')
  },
  {
    label: 'steak',
    value: 'steak',
    image: require('assets/images/steak.png')
  }
];
