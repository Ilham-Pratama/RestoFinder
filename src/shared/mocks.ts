export const BUSINESS_DUMMY = {
  id: 'WavvLdfdP6g8aZTtbBQHTw',
  alias: 'gary-danko-san-francisco',
  name: 'Gary Danko',
  image_url:
    'https://s3-media2.fl.yelpcdn.com/bphoto/CPc91bGzKBe95aM5edjhhQ/o.jpg',
  is_claimed: true,
  is_closed: false,
  url: 'https://www.yelp.com/biz/gary-danko-san-francisco?adjust_creative=wpr6gw4FnptTrk1CeT8POg&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_lookup&utm_source=wpr6gw4FnptTrk1CeT8POg',
  phone: '+14157492060',
  display_phone: '(415) 749-2060',
  review_count: 5296,
  categories: [
    {
      alias: 'newamerican',
      title: 'American (New)'
    },
    {
      alias: 'french',
      title: 'French'
    },
    {
      alias: 'wine_bars',
      title: 'Wine Bars'
    }
  ],
  rating: 4.5,
  location: {
    address1: '800 N Point St',
    address2: '',
    address3: '',
    city: 'San Francisco',
    zip_code: '94109',
    country: 'US',
    state: 'CA',
    display_address: ['800 N Point St', 'San Francisco, CA 94109'],
    cross_streets: ''
  },
  coordinates: {
    latitude: 37.80587,
    longitude: -122.42058
  },
  photos: [
    'https://s3-media2.fl.yelpcdn.com/bphoto/CPc91bGzKBe95aM5edjhhQ/o.jpg',
    'https://s3-media4.fl.yelpcdn.com/bphoto/FmXn6cYO1Mm03UNO5cbOqw/o.jpg',
    'https://s3-media4.fl.yelpcdn.com/bphoto/HZVDyYaghwPl2kVbvHuHjA/o.jpg'
  ],
  price: '$$$$',
  hours: [
    {
      open: [
        {
          is_overnight: false,
          start: '1730',
          end: '2200',
          day: 0
        },
        {
          is_overnight: false,
          start: '1730',
          end: '2200',
          day: 1
        },
        {
          is_overnight: false,
          start: '1730',
          end: '2200',
          day: 2
        },
        {
          is_overnight: false,
          start: '1730',
          end: '2200',
          day: 3
        },
        {
          is_overnight: false,
          start: '1730',
          end: '2200',
          day: 4
        },
        {
          is_overnight: false,
          start: '1730',
          end: '2200',
          day: 5
        },
        {
          is_overnight: false,
          start: '1730',
          end: '2200',
          day: 6
        }
      ],
      hours_type: 'REGULAR',
      is_open_now: false
    }
  ],
  transactions: [],
  special_hours: [
    {
      date: '2019-02-07',
      is_closed: null,
      start: '1600',
      end: '2000',
      is_overnight: false
    }
  ]
};

export const REVIEW_DUMMY = {
  id: 'xAG4O7l-t1ubbwVAlPnDKg',
  rating: 5,
  user: {
    id: 'W8UK02IDdRS2GL_66fuq6w',
    profile_url:
      'https://www.yelp.com/user_details?userid=W8UK02IDdRS2GL_66fuq6w',
    image_url:
      'https://s3-media3.fl.yelpcdn.com/photo/iwoAD12zkONZxJ94ChAaMg/o.jpg',
    name: 'Ella A.'
  },
  text: 'Went back again to this place since the last time i visited the bay area 5 months ago, and nothing has changed. Still the sketchy Mission, Still the cashier...',
  time_created: '2016-08-29 00:41:13',
  url: 'https://www.yelp.com/biz/la-palma-mexicatessen-san-francisco?hrid=hp8hAJ-AnlpqxCCu7kyCWA&adjust_creative=0sidDfoTIHle5vvHEBvF0w&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_reviews&utm_source=0sidDfoTIHle5vvHEBvF0w'
};

export const REVIEWS_DUMMY = {
  reviews: [
    {
      id: 'xAG4O7l-t1ubbwVAlPnDKg',
      rating: 5,
      user: {
        id: 'W8UK02IDdRS2GL_66fuq6w',
        profile_url:
          'https://www.yelp.com/user_details?userid=W8UK02IDdRS2GL_66fuq6w',
        image_url:
          'https://s3-media3.fl.yelpcdn.com/photo/iwoAD12zkONZxJ94ChAaMg/o.jpg',
        name: 'Ella A.'
      },
      text: 'Went back again to this place since the last time i visited the bay area 5 months ago, and nothing has changed. Still the sketchy Mission, Still the cashier...',
      time_created: '2016-08-29 00:41:13',
      url: 'https://www.yelp.com/biz/la-palma-mexicatessen-san-francisco?hrid=hp8hAJ-AnlpqxCCu7kyCWA&adjust_creative=0sidDfoTIHle5vvHEBvF0w&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_reviews&utm_source=0sidDfoTIHle5vvHEBvF0w'
    },
    {
      id: '1JNmYjJXr9ZbsfZUAgkeXQ',
      rating: 4,
      user: {
        id: 'rk-MwIUejOj6LWFkBwZ98Q',
        profile_url:
          'https://www.yelp.com/user_details?userid=rk-MwIUejOj6LWFkBwZ98Q',
        image_url: null,
        name: 'Yanni L.'
      },
      text: 'The "restaurant" is inside a small deli so there is no sit down area. Just grab and go.\n\nInside, they sell individually packaged ingredients so that you can...',
      time_created: '2016-09-28 08:55:29',
      url: 'https://www.yelp.com/biz/la-palma-mexicatessen-san-francisco?hrid=fj87uymFDJbq0Cy5hXTHIA&adjust_creative=0sidDfoTIHle5vvHEBvF0w&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_reviews&utm_source=0sidDfoTIHle5vvHEBvF0w'
    },
    {
      id: 'SIoiwwVRH6R2s2ipFfs4Ww',
      rating: 4,
      user: {
        id: 'rpOyqD_893cqmDAtJLbdog',
        profile_url:
          'https://www.yelp.com/user_details?userid=rpOyqD_893cqmDAtJLbdog',
        image_url: null,
        name: 'Suavecito M.'
      },
      text: 'Dear Mission District,\n\nI miss you and your many delicious late night food establishments and vibrant atmosphere.  I miss the way you sound and smell on a...',
      time_created: '2016-08-10 07:56:44',
      url: 'https://www.yelp.com/biz/la-palma-mexicatessen-san-francisco?hrid=m_tnQox9jqWeIrU87sN-IQ&adjust_creative=0sidDfoTIHle5vvHEBvF0w&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_reviews&utm_source=0sidDfoTIHle5vvHEBvF0w'
    }
  ],
  total: 3,
  possible_languages: ['en']
};
