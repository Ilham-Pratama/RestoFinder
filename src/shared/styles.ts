import { StyleSheet } from 'react-native';

export const ON_PRESS_BACKGROUND_COLOR = 'rgba(160, 160, 160, 0.2)';

const sharedStyles = StyleSheet.create({
  elevation: {
    shadowColor: 'black',
    shadowOffset: {
      height: 5,
      width: 5
    },
    elevation: 3,
    shadowOpacity: 0.1
  },

  flex1: {
    flex: 1
  },

  flexDirectionRow: {
    flexDirection: 'row'
  },

  alignItemsCenter: {
    alignItems: 'center'
  },

  justifyContentCenter: {
    justifyContent: 'center'
  },
  justifyContentSpaceBetween: {
    justifyContent: 'space-between'
  },

  padding1: {
    padding: 5
  },
  padding2: {
    padding: 10
  },
  padding3: {
    padding: 15
  },
  padding4: {
    padding: 20
  },

  paddingHorizontal1: {
    paddingHorizontal: 5
  },
  paddingHorizontal2: {
    paddingHorizontal: 10
  },
  paddingHorizontal3: {
    paddingHorizontal: 15
  },
  paddingHorizontal4: {
    paddingHorizontal: 20
  },

  paddingVertical1: {
    paddingVertical: 5
  },
  paddingVertical2: {
    paddingVertical: 10
  },
  paddingVertical3: {
    paddingVertical: 15
  },
  paddingVertical4: {
    paddingVertical: 20
  },

  marginTop1: {
    marginTop: 5
  },
  marginTop2: {
    marginTop: 10
  },
  marginTop3: {
    marginTop: 15
  },
  marginTop4: {
    marginTop: 20
  },

  marginBottom1: {
    marginBottom: 5
  },
  marginBottom2: {
    marginBottom: 10
  },
  marginBottom3: {
    marginBottom: 15
  },
  marginBottom4: {
    marginBottom: 20
  },

  marginRight1: {
    marginRight: 5
  },
  marginRight2: {
    marginRight: 10
  },
  marginRight3: {
    marginRight: 15
  },
  marginRight4: {
    marginRight: 20
  },

  marginHorizontal1: {
    marginHorizontal: 5
  },
  marginHorizontal2: {
    marginHorizontal: 10
  },
  marginHorizontal3: {
    marginHorizontal: 15
  },
  marginHorizontal4: {
    marginHorizontal: 20
  },

  marginVertical1: {
    marginVertical: 5
  },
  marginVertical2: {
    marginVertical: 10
  },
  marginVertical3: {
    marginVertical: 15
  },
  marginVertical4: {
    marginVertical: 20
  },

  textBold: {
    fontWeight: 'bold'
  },

  textSemiBold: {
    fontWeight: '700'
  }
});

export default sharedStyles;
