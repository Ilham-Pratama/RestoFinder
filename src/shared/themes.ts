import { Theme } from '@react-navigation/native';
import { DarkTheme } from '@react-navigation/native';

export const AppLightTheme: Theme = {
  dark: false,
  colors: {
    primary: '#c3f229',
    background: '#F2F2F2',
    card: '#FFF',
    text: '#1C1C1E',
    border: '#E2FFA7',
    notification: '#C1FC4C'
  }
};

export const AppDarkTheme: Theme = {
  dark: true,
  colors: {
    ...DarkTheme.colors,
    primary: '#3D3D3D',
    card: '#1A1A1A',
    background: '#121212'
  }
};
