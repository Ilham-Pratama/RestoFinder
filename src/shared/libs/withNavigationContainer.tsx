import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { AppLightTheme, AppDarkTheme } from 'shared/themes';

const withNavigationContainer = (component: any, darkMode: boolean = false) => {
  return () => (
    <NavigationContainer theme={darkMode ? AppDarkTheme : AppLightTheme}>
      {component}
    </NavigationContainer>
  );
};

export default withNavigationContainer;
