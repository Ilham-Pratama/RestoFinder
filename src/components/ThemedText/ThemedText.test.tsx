import React from 'react';
import { render } from '@testing-library/react-native';
import withNavigationContainer from 'shared/libs/withNavigationContainer';
import { AppDarkTheme, AppLightTheme } from 'shared/themes';
import ThemedText from '.';

describe('ThemedText', () => {
  const text = 'Sample Text';
  const defaultText = <ThemedText>{text}</ThemedText>;

  it('renders text in light mode', () => {
    const LightComponent = withNavigationContainer(defaultText);
    const lightComponent = render(<LightComponent />);

    expect(lightComponent.toJSON()).toMatchSnapshot();
    expect(lightComponent.getByText(text).props).toHaveProperty(
      'style.1.color',
      AppLightTheme.colors.text
    );
  });

  it('renders text in dark mode', () => {
    const DarkComponent = withNavigationContainer(defaultText, true);
    const darkComponent = render(<DarkComponent />);

    expect(darkComponent.toJSON()).toMatchSnapshot();
    expect(darkComponent.getByText(text).props).toHaveProperty(
      'style.1.color',
      AppDarkTheme.colors.text
    );
  });

  const color = '#121212';
  const darkModeColor = '#f4f4f4';
  const customText = (
    <ThemedText color={color} darkModeColor={darkModeColor}>
      {text}
    </ThemedText>
  );

  it('renders text in light mode with custom colors', () => {
    const LightComponent = withNavigationContainer(customText);
    const lightComponent = render(<LightComponent />);

    expect(lightComponent.toJSON()).toMatchSnapshot();
    expect(lightComponent.getByText(text).props).toHaveProperty(
      'style.1.color',
      color
    );
  });

  it('renders text in dark mode with custom colors', () => {
    const DarkComponent = withNavigationContainer(customText, true);
    const darkComponent = render(<DarkComponent />);

    expect(darkComponent.toJSON()).toMatchSnapshot();
    expect(darkComponent.getByText(text).props).toHaveProperty(
      'style.1.color',
      darkModeColor
    );
  });
});
