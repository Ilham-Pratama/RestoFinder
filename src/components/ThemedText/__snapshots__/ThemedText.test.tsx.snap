// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemedText renders text in dark mode 1`] = `
<Text
  style={
    Array [
      undefined,
      Object {
        "color": "rgb(229, 229, 231)",
      },
    ]
  }
>
  Sample Text
</Text>
`;

exports[`ThemedText renders text in dark mode with custom colors 1`] = `
<Text
  color="#121212"
  darkModeColor="#f4f4f4"
  style={
    Array [
      undefined,
      Object {
        "color": "#f4f4f4",
      },
    ]
  }
>
  Sample Text
</Text>
`;

exports[`ThemedText renders text in light mode 1`] = `
<Text
  style={
    Array [
      undefined,
      Object {
        "color": "#1C1C1E",
      },
    ]
  }
>
  Sample Text
</Text>
`;

exports[`ThemedText renders text in light mode with custom colors 1`] = `
<Text
  color="#121212"
  darkModeColor="#f4f4f4"
  style={
    Array [
      undefined,
      Object {
        "color": "#121212",
      },
    ]
  }
>
  Sample Text
</Text>
`;
